streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -53,7 +53,7 @@
  *     streams.start()
  *     ...
  *     final String queryableStoreName = table.queryableStoreName(); // returns null if KTable is not queryable
- *     ReadOnlyKeyValueStore view = streams.store(queryableStoreName, QueryableStoreTypes.keyValueStore());
+ *     TimestampedKeyValueStore view = streams.store(queryableStoreName, QueryableStoreTypes.keyValueStore());
```

Comment:
This change doesn't look quite right. It should still be a ReadOnlyKeyValueStore, but the generics are different: `ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>>`. Also, if the intent is to recommend using the Timestamped variant, then folks would actually have to specify `QueryableStoreTypes.timestampedKeyValueStore()`.

Reply:
This this top level class JavaDoc, should we also explain that it it still possible to query the store as plain key-value store only? I can imagine, that not all applications are interested in the timestamp and it would be quite convenient for them to stay with the old pattern (note, that the old pattern is not deprecated and still totally valid to use).

Thoughts?

Reply:
@vvcephei Thanks for the heads up. I've dug a little more and was able to confirm that streams.store returns a `ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>>` when `QueryableStoreTypes.timestampedKeyValueStore()` is passed as the second argument.  It looks like the type returned by streams.store is dependent upon the return type of the create method of that second argument, namely a [TimestampedKeyValueStoreType](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java#L128-L142). I'll make the necessary updates and verify that the code in the javadocs works. 

@mjsax You raise a good point. We can provide documentation for the two approaches; however, if we add another KeValueStore type then we've got to make more updates in many places. To remedy this, perhaps we only provide a code example of how to query at the KTable interface documentation and remove the examples in the documentation for the methods. Otherwise, we can continue to update these code snippets in the documentation throughout. What do you think?

Reply:
Well. I see your point. Maintaining the docs in many different places is quite a challenge. Curious to hear what @vvcephei thinks about this question.

Reply:
\cc @vvcephei Any comments?

Btw: There seems to be an overlap with https://issues.apache.org/jira/browse/KAFKA-9290 -- might be worth to resolve both issue with a single PR?